spi/common/core-spi/src/main/java/org/eclipse/edc/spi/asset/AssetIndex.java

```diff
@@ -83,11 +83,11 @@ public interface AssetIndex extends DataAddressResolver {
      * @param dataAddress The {@link DataAddress} to store
      * @return {@link StoreResult#success()} if the objects were stored, {@link StoreResult#alreadyExists(String)} when an object with the same ID already exists.
      */
-    default StoreResult<Void> accept(Asset asset, DataAddress dataAddress) {
-        return accept(new AssetEntry(asset, dataAddress));
+    default StoreResult<Void> updateOrCreate(Asset asset, DataAddress dataAddress) {
+        return updateOrCreate(new AssetEntry(asset, dataAddress));
     }
 
-    StoreResult<Void> accept(AssetEntry item);
+    StoreResult<Void> updateOrCreate(AssetEntry item);
```

Comment:
this is incorrect, it should just be `create()` as it does not follow upsert semantics.